### PR TITLE
Update CI build matrix for travis and appveyor

### PR DESCRIPTION
**Pull request by fs-eire**
- Use a fully customized matrix on travis
- Fix CMake missing on ubuntu precise
- Add AppVeyor support

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,47 @@
...
+    compiler: g++-5
+    addons:
+      apt:
+        sources:
+        - ubuntu-toolchain-r-test
+        packages:
+        - g++-5
+    env:
+    - COMPILER_OVERRIDE="CXX=g++-5 CC=gcc-5"
+  # Node LTS (6.x)      OS X (Yosemite)     AppleClang 6.1
```
Comment (daiyip) [inline .travis.yml]: AppleClang is actually LLVM.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "napajs",
-  "version": "0.1.0",
+  "version": "0.1.1",
```
Comment (daiyip) [inline package.json]: Not needed? since we don't change napajs distributables.

File: .travis.yml [hunk 3]
```diff
@@ -1,21 +1,47 @@
...
+  # Node 4.5.0          Linux (Precise)     G++5.4.1
+  - os: linux
+    dist: precise
+    node_js: '4.5.0'
+    compiler: g++-5
+    addons:
+      apt:
+        sources:
+        - ubuntu-toolchain-r-test
+        - george-edison55-precise-backports
```
Comment (daiyip) [inline .travis.yml]: What's this for?

Comment (fs-eire) [inline .travis.yml]: 'ubuntu-toolchain-r-test' is for g++5/g++6
'george-edison55-precise-backports' is for CMake 3.2+

Comment (daiyip) [inline .travis.yml]: Can you add comment for these two?

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+environment:
```
Comment (daiyip) [inline appveyor.yml]: please also add appveyor.yml to .npmignore.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+[![Build Status for Linux/MacOS](https://travis-ci.org/Microsoft/napajs.svg?branch=master)](https://travis-ci.org/Microsoft/napajs)
+[![Build Status for Windows](https://ci.appveyor.com/api/projects/status/github/Microsoft/napajs?branch=master&svg=true)](https://ci.appveyor.com/project/daiyip/napajs)
```
Comment (daiyip) [inline README.md]: Both icons are showing 'build', is it possible to customize the text on the icon?

Comment (daiyip) [inline README.md]: Given travis is not customizable, let's keep the two 'build' for now.

**Pull request merged**